espoon-voltti/evaka
# apigw: add new Suomi.fi prod certs, helper script

#### Summary

**NOTE:** This does not activate any of the new certificates, update to evaka-infra is required for that but that will be done when the new environment actually exists (after 2021-01-26).

- Add new Suomi.fi Tunnistus production certificates:
    1. saml-signing.idp.tunnistautuminen.suomi.fi.pem for the new production environment (will be activated on 2021-01-26, must transition before 2021-02-04)
    2. idp.tunnistaminen.suomi.fi.pem for the current environment (a temporary certificate intended for the transition period, we probably won't use this one)
- The migration to the new Suomi.fi production environment will start on 2021-01-26
- Create simple Python helper script to fetch SAML 2.0 IdP public certificates from their metadata URL
    - Extracts the public certificates from metadata XML and writes them in the correct format to files
        - Naming defaults to certificate's Common Name but can be overriden for silly Azure AD names ("Microsoft Azure Federated SSO Certificate")
    - Python selected due to ease-of-use for XML-walking and X509 parsing
- Document SAML metadata URLs for all current providers

#### Dependencies
None

#### Testing instructions
1. `cd apigw/config/certificates`
1. `pipenv install`
1. Follow documentation to fetch e.g. Suomi.fi production certificates with helper script
1. Nothing should change (but files should be updated with newer timestamps)

#### Checklist for pull request creator



#### Checklist for pull request reviewer (copy to review text box)


```
```
